src/js/ui/night/controls/brush-control.js

```diff
@@ -0,0 +1,445 @@
+    this._container.addEventListener('touchmove', this._onTouchMove)
+    this._container.addEventListener('mouseleave', this._onMouseLeave)
+  }
+
+  /**
+   * Bind event handlers
+   */
+  _bindEventHandler () {
+    this._onMouseDown = this._onMouseDown.bind(this)
+    this._onMouseUp = this._onMouseUp.bind(this)
+    this._onMouseMove = this._onMouseMove.bind(this)
+    this._onMouseLeave = this._onMouseLeave.bind(this)
+    this._onTouchStart = this._onTouchStart.bind(this)
+    this._onTouchEnd = this._onTouchEnd.bind(this)
+    this._onTouchMove = this._onTouchMove.bind(this)
+  }
+
+  /**
+   * Sets up the slider used to change the brush size
+   */
+  _setupSlider () {
+    let sliderElement = this._controls.querySelector('.imglykit-slider')
+    this._slider = new SimpleSlider(sliderElement, {
+      minValue: 1,
+      maxValue: 30
+    })
+    this._onThicknessUpdate = this._onThicknessUpdate.bind(this)
+    this._slider.on('update', this._onThicknessUpdate)
+    this._slider.setValue(this._currentThickness * this._getLongerSideSize())
+  }
+
+  /**
+   * Sets up the color picker used to change the brush color
+   */
+  _setupColorPicker () {
+    let colorPickerElement = this._controls.querySelector('.imglykit-color-picker')
+    this._colorPicker = new ColorPicker(this._ui, colorPickerElement)
+    this._colorPicker.on('update', this._onColorUpdate.bind(this))
+    this._colorPicker.setValue(this._currentColor)
+  }
+
+  /**
+   * Gets called when the back button has been clicked
+   * @override
+   */
+  _onBack () {
+    this._ui.canvas.setZoomLevel(this._initialZoomLevel, false)
+    if (this._operationExistedBefore) {
+      this._resetOperationSettings()
+    } else {
+      this._ui.removeOperation('brush')
+    }
+    this._ui.canvas.render()
+  }
+
+  /**
+   * Resets the operation options to the initial options
+   */
+  _resetOperationSettings () {
+    console.log (this._initialOptions)
+    this._operation.setControlPoints(this._initialOptions.controlPoints)
+    this._operation.setButtonStatus(this._initialOptions.buttonStatus)
+    this._operation.setColors(this._initialOptions.colors)
+    this._operation.setThicknesses(this._initialOptions.thicknesses)
+  }
+
+  /**
+   * Gets called when the done button has been clicked
+   * @protected
+   */
+  _onDone () {
+    this._ui.canvas.setZoomLevel(this._initialZoomLevel, false)
+    this._ui.addHistory(this, {
+      colors: this._initialOptions.colors,
+      thicknesses: this._initialOptions.thicknesses,
+      controlPoints: this._initialOptions.controlPoints,
+      buttonStatus: this._initialOptions.buttonStatus
+    }, this._operationExistedBefore)
+  }
+
+  /**
+   * The data that is available to the template
+   * @type {Object}
+   * @override
+   */
+  get context () {
+    let context = super.context
+    // Extend context like this...
+    context.myVariable = 'myValue'
+    return context
+  }
+
+  _onTouchStart (e) {
```

Comment:
nope they act differently due the custom cursor.
When the user moves the mouse without painting you show the and move the cursor.
You don't do that on touch events.

Please read the method-'pairs' so you get the idea